crates/bevy_asset/src/lib.rs

```diff
@@ -283,6 +283,14 @@ pub trait AssetApp {
     fn register_asset_loader<L: AssetLoader>(&mut self, loader: L) -> &mut Self;
     /// Registers the given `processor` in the [`App`]'s [`AssetProcessor`].
     fn register_asset_processor<P: Process>(&mut self, processor: P) -> &mut Self;
+    /// Registers the given `processor` in the [`App`]'s [`AssetProcessor`] along with an extra alias.
+    ///
+    /// This alias can be used in meta files to refer to this asset processor without using the full type name.
+    fn register_asset_processor_with_alias<P: Process>(
```

Comment:
Still on vacation so this is just a quick review: I think this approach is solid, especially in the short term. I do think migrating to TypePath + supporting short names is the right long term move (this code was written in a pre-TypePath-merge world), but even in that case, it doesn't fully solve the "generic flattening / LoadAndSave erasure problem". We'd need to use that in combination with something like what @JMS55 suggested awhile back on discord (define a new short type name that shims `LoadAndSave<X, Y>`). Rather than discuss whether or not that is the "right" final approach here, we should just merge  "aliases" as they are simple and they get the job done.

Random musings: to support the more general "one to many asset transformations" space, we'll need to make meta serialization more type erased + dynamic than it currently is. That might also allow us to rethink how thing like LoadAndSave are expressed.